server/routes/login.js

```diff
@@ -15,165 +16,161 @@ const notify = url ? new NotifyClient(url, clientKey) : new NotifyClient(clientK
 const notifySmsTemplate = smsTemplateId || ''
 const notifyEmailTemplate = emailTemplateId || ''
 
-module.exports = () => (router) => {
-  router.get(
-    '/',
-    asyncMiddleware(async (req, res) => {
-      postLogin(req, res)
-    }),
-  )
-
-  router.get(
-    '/auth/login',
-    asyncMiddleware(async (req, res) => {
-      postLogin(req, res)
-    }),
-  )
-
-  router.post(
-    '/auth/2fa',
-    asyncMiddleware(async (req, res) => {
-      const userAuthenticationDetails = await userAuthenticationService.getUserAuthenticationDetails(req.user.username)
-
-      const inputTwoFactorCode = utilities.createTwoFactorAuthenticationHash(req.body.code)
-
-      if (inputTwoFactorCode === userAuthenticationDetails[0].TwoFactorAuthenticationHash) {
-        req.user.employeeName = jwtDecode(req.user.token).name
-
-        await userAuthenticationService.updateUserSessionExpiryAndLastLoginDateTime(
-          req.user.username,
-          new Date(Date.now() + config.hmppsCookie.expiryMinutes * 60 * 1000),
-        )
-
-        res.redirect(`/calendar/${utilities.getStartMonth()}`)
-      } else {
-        logError(req.url, '2FA failure')
-        res.render('pages/two-factor-auth', { authError: true, csrfToken: res.locals.csrfToken })
-      }
-    }),
-  )
-
-  const postLogin = asyncMiddleware(async (req, res) => {
-    // if maintenance start/end dates exist then dcheck whether to display maintenance page
-    // otherwise just ignore the following, it will become effective as soon as those environment
-    // variables are created.  13DEC19.
-    try {
-      if (!utilities.isNullOrEmpty(config.maintenance.start) && !utilities.isNullOrEmpty(config.maintenance.end)) {
-        // eslint-disable-next-line vars-on-top
-        const maintenanceStartDateTime = Date.parse(config.maintenance.start)
-          ? new Date(config.maintenance.start)
-          : null
-        const maintenanceEndDateTime = Date.parse(config.maintenance.end) ? new Date(config.maintenance.end) : null
-
-        if (utilities.calculateMaintenanceDates(maintenanceStartDateTime, maintenanceEndDateTime)) {
-          res.render('pages/maintenance', {
-            startDateTime: maintenanceStartDateTime,
-            endDateTime: maintenanceEndDateTime,
-            csrfToken: res.locals.csrfToken,
-          })
-          return
-        }
+router.get(
+  '/',
+  asyncMiddleware(async (req, res) => {
+    postLogin(req, res)
+  }),
+)
+
+router.get(
+  '/auth/login',
+  asyncMiddleware(async (req, res) => {
+    postLogin(req, res)
+  }),
+)
+
+router.post(
+  '/auth/2fa',
+  asyncMiddleware(async (req, res) => {
+    const userAuthenticationDetails = await userAuthenticationService.getUserAuthenticationDetails(req.user.username)
+
+    const inputTwoFactorCode = utilities.createTwoFactorAuthenticationHash(req.body.code)
+
+    if (inputTwoFactorCode === userAuthenticationDetails[0].TwoFactorAuthenticationHash) {
+      req.user.employeeName = jwtDecode(req.user.token).name
```

Comment:
Is it just me or does adding to the request mid handler feel weird

Reply:
Out of scope for this PR. IMHO all these routes need rewriting with a ticket/commit each. I think there is also a case for utilising `res.locals` in all our middleware too. But I've tried to touch as little of the inner workings as possible in this MR. Its literally to remove obfuscation from the the routes and `app.js`